src/Extensions/Features/test/FeatureCollectionTests.cs

```diff
@@ -44,4 +45,62 @@ public void SetNullValueRemoves()
         var thing2 = interfaces[typeof(IThing)];
         Assert.Null(thing2);
     }
+
+    [Fact]
+    public void GetMissingStructFeatureThrows()
+    {
+        var interfaces = new FeatureCollection();
+
+        // Regression test: Used to throw NullReferenceException because it tried to unbox a null object to a struct
+        var ex = Assert.Throws<InvalidOperationException>(() => interfaces.Get<int>());
+        Assert.Equal("System.Int32 does not exist in the feature collection and because it is a struct we can't return null. Use 'featureCollection[typeof(System.Int32)] is not null' to check if it exists.", ex.Message);
```

Comment:
```suggestion
        Assert.Equal("System.Int32 does not exist in the feature collection and because it is a struct the method can't return null. Use 'featureCollection[typeof(System.Int32)] is not null' to check if the feature exists.", ex.Message);
```